src/flightlog_fielddefs.js

```diff
@@ -488,53 +531,59 @@ export function adjustFieldDefsList(firmwareType, firmwareVersion) {
     DEBUG_MODE = makeReadOnly(DEBUG_MODE);
 
     // Flight mode names
-    FLIGHT_LOG_FLIGHT_MODE_NAME = FLIGHT_LOG_FLIGHT_MODE_NAME_POST_3_3.slice(0);
-    if (semver.lt(firmwareVersion, "3.4.0")) {
-      FLIGHT_LOG_FLIGHT_MODE_NAME.splice(
-        FLIGHT_LOG_FLIGHT_MODE_NAME.indexOf("GPSRESCUE"),
-        1
-      );
-    }
-    if (semver.gte(firmwareVersion, "3.5.0")) {
-      FLIGHT_LOG_FLIGHT_MODE_NAME.splice(
-        FLIGHT_LOG_FLIGHT_MODE_NAME.indexOf("RANGEFINDER"),
-        1
-      );
-      FLIGHT_LOG_FLIGHT_MODE_NAME.splice(
-        FLIGHT_LOG_FLIGHT_MODE_NAME.indexOf("CAMTRIG"),
-        1
-      );
-      FLIGHT_LOG_FLIGHT_MODE_NAME.splice(
-        FLIGHT_LOG_FLIGHT_MODE_NAME.indexOf("LEDMAX"),
-        1
-      );
-      FLIGHT_LOG_FLIGHT_MODE_NAME.splice(
-        FLIGHT_LOG_FLIGHT_MODE_NAME.indexOf("LLIGHTS"),
-        1
-      );
-      FLIGHT_LOG_FLIGHT_MODE_NAME.splice(
-        FLIGHT_LOG_FLIGHT_MODE_NAME.indexOf("GOV"),
-        1
-      );
-      FLIGHT_LOG_FLIGHT_MODE_NAME.splice(
-        FLIGHT_LOG_FLIGHT_MODE_NAME.indexOf("GTUNE"),
-        1
-      );
+    if (semver.gte(firmwareVersion, "4.6.0")) {
+      FLIGHT_LOG_FLIGHT_MODE_NAME = FLIGHT_LOG_FLIGHT_MODE_NAME_POST_4_6.slice(0);
     }
-    if (semver.gte(firmwareVersion, "4.0.0")) {
-      FLIGHT_LOG_FLIGHT_MODE_NAME.splice(
-        FLIGHT_LOG_FLIGHT_MODE_NAME.indexOf("BARO"),
-        1
-      );
-      FLIGHT_LOG_FLIGHT_MODE_NAME.splice(
-        FLIGHT_LOG_FLIGHT_MODE_NAME.indexOf("GPSHOME"),
-        1
-      );
-      FLIGHT_LOG_FLIGHT_MODE_NAME.splice(
-        FLIGHT_LOG_FLIGHT_MODE_NAME.indexOf("GPSHOLD"),
-        1
-      );
+    else {
```

Comment:
This else belongs to 536 ?

Think code in 534 - 536 should go to 586 to follow the version history.

Reply:
534: if (semver.gte(firmwareVersion, "4.6.0")) {
      FLIGHT_LOG_FLIGHT_MODE_NAME = FLIGHT_LOG_FLIGHT_MODE_NAME_POST_4_6.slice(0);
    }
    else {
            the olde code
    }

I am not seeing error.
The code style improvement  to follow versions history?

Reply:
There is no error - but with the else the other part is not executed. It makes changes dependent on firmware version.

IIRC code patches additions / removals depending on firmware version. Which should go from old to new.

```
if (gte 3.5) make changes
if (gte 3.6) make more changes
...
if (gte 4.6) make latest changes
```

Reply:
I can not do these actions in case of 4.6
if (gte 3.5) make changes
if (gte 3.6) make more changes
......

Reply:
Somehow not able to mark this line for deletion.

Reply:
Missed the part you made a new list implementing those changes. See review below.